daemon/cmd/kube_proxy_replacement.go

```diff
@@ -448,6 +415,43 @@ func finishKubeProxyReplacementInit(isKubeProxyReplacementStrict bool) {
 
 	// After this point, BPF NodePort should not be disabled
 
+	if !option.Config.EnableHostLegacyRouting {
+		msg := ""
+		switch {
+		case !option.Config.EnableNodePort:
+			msg = fmt.Sprintf("BPF host routing requires %s.", option.EnableNodePort)
+		case option.Config.Tunnel != option.TunnelDisabled:
+			msg = fmt.Sprintf("BPF host routing is only available in native routing mode.")
+		// Needs host stack for packet handling.
+		case option.Config.EnableEndpointRoutes:
+			msg = fmt.Sprintf("BPF host routing is incompatible with %s.", option.EnableEndpointRoutes)
+		case option.Config.EnableIPSec:
+			msg = fmt.Sprintf("BPF host routing is incompatible with %s.", option.EnableIPSecName)
+		// Non-BPF masquerade requires netfilter and hence CT.
+		case (option.Config.EnableIPv4Masquerade || option.Config.EnableIPv6Masquerade) &&
+			!option.Config.EnableBPFMasquerade:
+			msg = fmt.Sprintf("BPF host routing requires %s.", option.EnableBPFMasquerade)
+		case !hasHardwareAddress():
+			msg = fmt.Sprintf("BPF host routing is incompatible with devices without L2 addr.")
```

Comment:
Could you elaborate on why incompatible? I think it should be working just that we need to push a L2 header before the redirect_peer? Although I'm not even sure if the stack cares about that even. Maybe needs the same change as in https://github.com/cilium/cilium/pull/15148 to get it working?

Reply:
> I think it should be working just that we need to push a L2 header before the redirect_peer?

As far as I've checked the code path, it should be enough. I didn't want to add to this PR to avoid making it too complex.

Reply:
Ok, if it's follow-up, lets just say `currently not supported` rather than `incompatible`. Hopefully for 1.10 scope, we'll resolve it in time. I can also help if needed.